File: javascript/004.js
/**
 * Project Euler - Problem Solution 004
 * Copyright (c) Justin McGettigan. All rights reserved.
 * https://github.com/jwmcgettigan/project-euler-solutions
 */

const isPalindrome = (product) => (
  String(product) === String(product).split('').reverse().join('')
);

const largestPalindromeOfProducts = (lower_limit, upper_limit) => {
  let products = [];
  for(let x = lower_limit; x < upper_limit ; x++) {
    for(let y = x; y < upper_limit; y++) {
      let product = x * y;
      if(isPalindrome(product)) {
        products.push(product);
      }
    }
  }
  return Math.max(...products)
}

console.log(largestPalindromeOfProducts(100, 1000));

File: python/004.py
"""
Project Euler - Problem Solution 004
Copyright (c) Justin McGettigan. All rights reserved.
https://github.com/jwmcgettigan/project-euler-solutions
"""

def is_palindrome(product) -> bool:
  return str(product) == str(product)[::-1]

def largest_palindrome_of_products(lower_limit, upper_limit) -> int:
  products = []
  for x in range(lower_limit, upper_limit):
    for y in range(x, upper_limit):
      product = x * y
      if(is_palindrome(product)):
        products.append(product)
  return max(products)

if __name__ == "__main__":
	print(largest_palindrome_of_products(100, 1000))